src/lambda/wizards/samDeployWizard.ts

```diff
@@ -340,14 +343,45 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
     }
 
     /**
-     * Retrieves an S3 Bucket to deploy to from the user.
+     * Prompts user to select a bucket from a regionalized list of S3 buckets
+     * Caches buckets for the entire Wizard run the first time it hits this step
+     * Allows the user to go back a step if the selected region doesn't have any buckets
      *
+     * @param selectedRegion Region selected in the last step. Used to focus S3 bucket list
      * @param initialValue Optional, Initial value to prompt with
+     * @param s3client Optional, S3 client (defaults to DefaultS3Client)
      *
-     * @returns S3 Bucket name. Undefined represents cancel.
+     * @returns S3 Bucket name. Undefined represents back.
      */
-    public async promptUserForS3Bucket(selectedRegion: string, initialValue?: string): Promise<string | undefined> {
-        const inputBox = input.createInputBox({
+    public async promptUserForS3Bucket(
+        selectedRegion: string,
+        initialValue?: string,
+        s3Client: S3Client = new DefaultS3Client(selectedRegion)
```

Comment:
The S3 client should be provided to `DefaultSamDeployWizardContext`. This method is defined in SamDeployWizardContext and adding an arg here doesn't seem right.

Reply:
The tough part is that this relies on the `selectedRegion` to instantiate the client, which is only available to this step.

Right now, we don't have a way to test this file (due to nature of the quick pick). I'm thinking of moving this out of the args and into the function itself?

Reply:
✔️ to my suggested implementation.